server/middlewares/UserValidator.js

```diff
@@ -0,0 +1,128 @@
+import expressValidator from 'express-validator/check';
+import helpers from '../helpers';
+import checkForErrors from './checkForErrors';
+
+const { check } = expressValidator;
+const { capitaliseFirstLetter, makeLowerCase } = helpers;
+
+/**
+ * @class UserValidator
+ * @classdesc Provides validation middlewares for login and signup route
+ */
+export default class UserValidator {
+  /**
+  * Email validator
+  * @returns {function} call to a Check API middleware
+  * @memberof Validation
+  */
+  static checkEmail() {
+    return check('email')
+      .exists().withMessage('Email is missing')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('Email cannot be blank')
+      .isEmail()
+      .withMessage('Email is not valid')
+      .blacklist(' ')
+      .customSanitizer(value => makeLowerCase(value));
+  }
+
+  /**
+  * Firstname validator
+  * @returns {function} call to a Check API middleware
+  * @memberof Validation
+  */
+  static checkFirstName() {
+    return check('firstName')
+      .exists().withMessage('Firstname is missing')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('Firstname cannot be blank')
+      .isAlpha()
+      .withMessage('Firstname can only contain alphabets')
+      .isLength({ min: 2, max: 20 })
+      .withMessage('Firstname must be at least 2 characters, and maximum 20')
+      .blacklist(' ')
+      .customSanitizer(value => capitaliseFirstLetter(value));
+  }
+
+  /**
+  * Lastname validator
+  * @returns {function} call to a Check API middleware
+  * @memberof Validation
+  */
+  static checkLastName() {
+    return check('lastName')
+      .exists().withMessage('Lastname is missing')
```

Comment:
What if you create a function for name check and call the function for both first and last name while passing the type of name as a parameter?